### ISSUE TITLE
[BUG] physical device not online (The reconnection feature)

### ISSUE THREAD

**Issue by Datak3** [Baldhor/Homey-ESPhome-Enhanced#88]
**Describe the bug**

**To Reproduce**
1. Go to 'Front page opf homey where the device are'
2. Click on 'the device thats is connected with ESPhome'
3. Scroll down to 'on the logo you will see a error mark ! in red'

**Expected behavior**

**Screenshots**
Not at this moment but see the homey board/forum for the error logs and so on

**Additional context**
You can use the "collect debug info" feature to attach relevant data to the issue: [User Guide](https://github.com/Baldhor/Homey-ESPhome-Enhanced/wiki/User-Guide#manage-your-physical-devices-page)

**Comment by Baldhor** [comment]
Issue opened by @Patrick_85 (his name on the Homey community).

It concerns the device: https://www.hobbyelectronica.nl/product/air-conditioner-wi-fi-module-slwf-01-pro/?gclid=EAIaIQobChMI8trZtMatgwMV-5aDBx1RVQYPEAQYBiABEgLSm_D_BwE

It's default configuration is:
```
esphome:
  name: air-conditioner
  platform: ESP8266
  board: esp12e
  name_add_mac_suffix: true

wifi:

  # Enable fallback hotspot (captive portal) in case wifi connection fails
  ap:
    ssid: "AC-wifi"
    password: "slwf01pro"

captive_portal:

# Enable logging
logger:
  baud_rate: 0

# Enable Home Assistant API
api:

ota:

uart:
  tx_pin: 1
  rx_pin: 3
  baud_rate: 9600
  
  
climate:
  - platform: midea
    name: Midea Climate         # Use a unique name.
    period: 1s                  # Optional
    timeout: 2s                 # Optional
    num_attempts: 3             # Optional
    autoconf: true              # Autoconfigure most options.
    beeper: true                # Beep on commands.
    visual:                     # Optional. Example of visual settings override.
      min_temperature: 17 °C    # min: 17
      max_temperature: 30 °C    # max: 30
      temperature_step: 0.5 °C  # min: 0.5
    supported_modes:            # All capabilities in this section detected by autoconf.
      - FAN_ONLY                # This capability is always used.
      - HEAT_COOL
      - COOL
      - HEAT
      - DRY
    custom_fan_modes:
      - SILENT
      - TURBO
    supported_presets:          # All capabilities in this section detected by autoconf.
      - ECO
      - BOOST
      - SLEEP                   # This capability is always used.
    custom_presets:             # All capabilities in this section detected by autoconf.
      - FREEZE_PROTECTION
    supported_swing_modes:
      - VERTICAL                # This capability is always used.
      - HORIZONTAL
      - BOTH
    outdoor_temperature:        # Optional. Outdoor temperature sensor (may display incorrect values after long inactivity).
      name: Temp
    power_usage:                # Optional. Power usage sensor (only for devices that support this feature).
      name: Power
    humidity_setpoint:          # Optional. Indoor humidity sensor (only for devices that support this feature).
      name: Humidity


web_server:
  port: 80
```

**Comment by Baldhor** [comment]
![image](https://github.com/Baldhor/Homey-ESPhome-Enhanced/assets/57640917/c71ed456-e883-4101-ac45-ba3a10201726)

@Datak3 can you check the health of the homey application at that moment using homey graphics?

This link should be the right one: https://my.homey.app/homeys/5d14b4961b89b60c8f9df337/insights?logs1=homey:manager:apps:nl.inversion.esphome-cpuusage&logs2=homey:manager:apps:nl.inversion.esphome-memusage

Make sure to use the right timeframe to cover the moment of the logs:
![image](https://github.com/Baldhor/Homey-ESPhome-Enhanced/assets/57640917/3356e988-fd7b-49dc-906b-3724e91a16db)


Strange enough, both your device "disconnected" at the same time ...

**Comment by Baldhor** [comment]
and just before they disconnected, they sent lot of logs (more than usually) ... very strange

**Comment by Baldhor** [comment]
@Datak3 

![image](https://github.com/Baldhor/Homey-ESPhome-Enhanced/assets/57640917/6292cd84-18ed-455b-83bd-05c319b0395e)

I bet the firmware of your devices is based on an old version of ESPhome ... in the meantime, there was many fixes!
So I recompiled the yml file and included the binary with latest version of ESPhome in my git ...
=> https://github.com/Baldhor/Homey-ESPhome-Enhanced/tree/main/esphome-binary/midea-air-conditioner

**Comment by Datak3** [comment]
Hi Baldhor,If I understand you right. For now I need to update the device with the bin doc you did send me ?If it works don’t forget to send me a link for your contribution. Best regards,Met vriendelijke groet,Patrick Tak. Op 20 feb 2024 om 01:23 heeft Baldhor ***@***.***> het volgende geschreven:﻿
@Datak3
image.png (view on web)
I bet the firmware of your devices is based on an old version of ESPhome ... in the meantime, there was many fixes!
So I recompiled the yml file and included the binary with latest version of ESPhome in my git ...
=> https://github.com/Baldhor/Homey-ESPhome-Enhanced/tree/main/esphome-binary/midea-air-conditioner

**Comment by Baldhor** [comment]
Avoir updating both device, even if it work. The behaviour of the app is
Stiller wrong, restarting it is anormal,

**Comment by Datak3** [comment]
Hi,

 

I did the update the device is not reacable and not findable on my wifi so I thinks its bricked or the update went wrong.

I wont update the other one before I get this one fixed.

 

 

 

Van: Baldhor ***@***.***> 
Verzonden: dinsdag 20 februari 2024 08:47
Aan: Baldhor/Homey-ESPhome-Enhanced ***@***.***>
CC: Datak3 ***@***.***>; Mention ***@***.***>
Onderwerp: Re: [Baldhor/Homey-ESPhome-Enhanced] [BUG] physical device not online (The reconnection feature) (Issue #88)

 

Avoir updating both device, even if it work. The behaviour of the app is 
Stiller wrong, restarting it is anormal,

**Comment by Baldhor** [comment]
> I did the update the device is not reacable and not findable on my wifi so I thinks its bricked or the update went wrong. I wont update the other one before I get this one fixed.

if nothing connect to your device, it will restart automatically after a little while (5 minutes if I remember).
Let's wait a little and be confident :)

**Comment by Baldhor** [comment]
@Datak3 still activate console.re please

**Comment by Baldhor** [comment]
I still post was I was working on before you met that brick issue ...

@Datak3 let's explain clearly and step by step :)

1/ Your device has an old firmware
=> I believe it's the main reason of your issue
=> So I retrieved the config of your device and compiled it with the latest version
=> Install it on ONLY one device please (see next steps :) )

2/ The device with the new firmware should "I hope" works as expected
=> Not the old one, as it still has the old firmware

3/ Even the old firmware as a problem, the autoreconnect feature should works!
=> So I prepared a new fix for my app ... https://homey.app/a/nl.inversion.esphome/test/
=> **Install it after step 2 please** (not before)

4/ Hopefully, this new fix should make both your deviced behave correctly
=> Please confirm it

5/ You can finaly update your second device
=> An up to date firmware is always better ...


**About the new fix:**
The low level api I use has an embeded auto reconnect feature, it's like a chirurgical strike ... which is very eco friendly.
The fix I made put back the old implement I done few months ago, it's more like a "nuclear bomb" :)
It's not eco friendly, as it just delete everything and redo everything :)

**Comment by Baldhor** [comment]
spaming, ... but I bet your device changed its IP address !
because I renamed the device from air-conditionner to midea-air-conditioner

If it doesn't connect to your wifi, it will expose  an AP hotspot:
    ssid: "AC-wifi"
    password: "slwf01pro"

**Comment by Baldhor** [comment]
all good for now then, let's see if the updated device behave better during the next hours ...

**Comment by Datak3** [comment]
No nooo its no spam 😊

 

Van: Baldhor ***@***.***> 
Verzonden: dinsdag 20 februari 2024 19:32
Aan: Baldhor/Homey-ESPhome-Enhanced ***@***.***>
CC: Datak3 ***@***.***>; Mention ***@***.***>
Onderwerp: Re: [Baldhor/Homey-ESPhome-Enhanced] [BUG] physical device not online (The reconnection feature) (Issue #88)

 

spaming, ... but I bet your device changed its IP address !
because I renamed the device from air-conditionner to midea-air-conditioner

If it doesn't connect to your wifi, it will expose an AP hotspot:
ssid: "AC-wifi"
password: "slwf01pro"

**Comment by Datak3** [comment]
Yes, Al good for now. I will update the other one in a couple of hours. Met vriendelijke groet,Patrick Tak. Mob: 06 50 59 59 01www.patrick-tak.nlOp 20 feb 2024 om 19:45 heeft Baldhor ***@***.***> het volgende geschreven:﻿
all good for now then, let's see if the updated device behave better during the next hours ...

**Comment by Baldhor** [comment]
> Yes, Al good for now. I will update the other one in a couple of hours. Met vriendelijke groet,Patrick Tak. Mob: 06 50 59 59 01www.patrick-tak.nlOp 20 feb 2024 om 19:45 heeft Baldhor ***@***.***> het volgende geschreven:﻿ all good for now then, let's see if the updated device behave better during the next hours ... —Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you were mentioned.Message ID: ***@***.***>

NOOOOOOOOOOOOOOOOOOOOOO

Read my post: https://github.com/Baldhor/Homey-ESPhome-Enhanced/issues/88#issuecomment-1954822447

**Comment by Baldhor** [comment]
> Okeeeee I do not update. 🙈To bad baldhor it wantje one I updated 😞Op 20 feb 2024 om 20:16 heeft Baldhor ***@***.***> het volgende geschreven:﻿ @Datak3 make sure you read previous post please :) Seems aircoslaapkamer is dead already :) image.png (view on web) Tell me it's the one you DIT NOT updated :) —Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you were mentioned.Message ID: ***@***.***>

ok then updating the firmware do not solve the issue, you can install the test version of my app then :)

**Comment by Datak3** [comment]
I also did the update on homey “app”And this did a reset on the app. We just wait now, so see if it did work with your previous changes on disconnecting. Op 20 feb 2024 om 20:16 heeft Baldhor ***@***.***> het volgende geschreven:﻿
@Datak3 make sure you read previous post please :)
Seems aircoslaapkamer is dead already :)
image.png (view on web)
Tell me it's the one you DIT NOT updated :)

**Comment by Datak3** [comment]
Just did 💪🏻 Op 20 feb 2024 om 20:30 heeft Baldhor ***@***.***> het volgende geschreven:﻿

Okeeeee I do not update. 🙈To bad baldhor it wantje one I updated 😞Op 20 feb 2024 om 20:16 heeft Baldhor @.> het volgende geschreven:﻿ @Datak3 make sure you read previous post please :) Seems aircoslaapkamer is dead already :) image.png (view on web) Tell me it's the one you DIT NOT updated :) —Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you were mentioned.Message ID: @.>

ok then updating the firmware do not solve the issue, you can install the test version of my app then :)

**Comment by Baldhor** [comment]
I published a new test version, the auto reconnect feature was not always working ... it should be better now

**Comment by Datak3** [comment]
Hi, I had no problems sinds the last update. “From last evening” even with the firmware update. So both devices did work oké from last night. As of now i did the update to 1.3.8I’m gonna test the update 😄Op 21 feb 2024 om 19:12 heeft Baldhor ***@***.***> het volgende geschreven:﻿
I published a new test version, the auto reconnect feature was not always working ... it should be better now

**Comment by Baldhor** [comment]
> Hi, I had no problems sinds the last update. “From last evening” even with the firmware update. So both devices did work oké from last night. As of now i did the update to 1.3.8I’m gonna test the update 😄Op 21 feb 2024 om 19:12 heeft Baldhor ***@***.***> het volgende geschreven:﻿ I published a new test version, the auto reconnect feature was not always working ... it should be better now —Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you were mentioned.Message ID: ***@***.***>

good for you, at 4pm today I found all my ESPhome device disconnected lol

**Comment by Datak3** [comment]
Ahw crap your kidding me. NooooooOp 21 feb 2024 om 21:37 heeft Baldhor ***@***.***> het volgende geschreven:﻿

Hi, I had no problems sinds the last update. “From last evening” even with the firmware update. So both devices did work oké from last night. As of now i did the update to 1.3.8I’m gonna test the update 😄Op 21 feb 2024 om 19:12 heeft Baldhor @.> het volgende geschreven:﻿ I published a new test version, the auto reconnect feature was not always working ... it should be better now —Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you were mentioned.Message ID: @.>

good for you, at 4pm today I found all my ESPhome device disconnected lol

**Comment by Baldhor** [comment]
seems to be working out fine, so I published as live version

**Comment by Datak3** [comment]
Yeah for me also, no connection loss. Sometimes it can take a couple of days but I’m hopeful now 😄Op 22 feb 2024 om 21:20 heeft Baldhor ***@***.***> het volgende geschreven:﻿
seems to be working out fine, so I published as live version

**Comment by Baldhor** [comment]
I will consider it solved for now, please reopen if needed

**Comment by Datak3** [comment]
Will do thank you so far 😎Op 23 feb 2024 om 12:00 heeft Baldhor ***@***.***> het volgende geschreven:﻿
Closed #88 as completed.
